frontend/src/component/changeRequest/ChangeRequest/Changes/Change/StrategyChange.tsx

```diff
@@ -127,20 +127,19 @@ export const StrategyChange: VFC<{
         environmentName,
     );
 
+    const hasDiff = (object: unknown, objectToCompare: unknown) =>
+        JSON.stringify(object) !== JSON.stringify(objectToCompare);
+
     const isStrategyAction =
         change.action === 'addStrategy' || change.action === 'updateStrategy';
 
-    const featureStrategyVariantsDisplay =
+    const hasVariantDiff =
         isStrategyAction &&
+        currentStrategy?.variants &&
+        currentStrategy?.variants?.length > 0 &&
         change.payload.variants &&
-        change.payload.variants.length > 0 ? (
-            <StyledBox>
-                <StyledTypography>
-                    Updating feature variants to:
-                </StyledTypography>
-                <EnvironmentVariantsTable variants={change.payload.variants} />
-            </StyledBox>
-        ) : null;
+        change.payload.variants.length > 0 &&
```

Comment:
Will this fail if the new strategy has no variants but the old one did?

Reply:
Good catch

Reply:
Fixed